File: pdu/python/hako_srv_msgs/pdu_cdr_conv_SimControlResponsePacket.py
from .pdu_pytype_SimControlResponsePacket import SimControlResponsePacket
from ..pdu_cdr_runtime import CdrReader, CdrWriter

from ..hako_srv_msgs.pdu_cdr_conv_ServiceResponseHeader import *
from ..hako_srv_msgs.pdu_cdr_conv_SimControlResponse import *



def py_to_cdr_body_SimControlResponsePacket(writer: CdrWriter, src: SimControlResponsePacket):
    py_to_cdr_body_ServiceResponseHeader(writer, src.header)
    py_to_cdr_body_SimControlResponse(writer, src.body)


def cdr_body_to_py_SimControlResponsePacket(reader: CdrReader, dst: SimControlResponsePacket):
    cdr_body_to_py_ServiceResponseHeader(reader, dst.header)
    cdr_body_to_py_SimControlResponse(reader, dst.body)
    return dst


def py_to_cdr_SimControlResponsePacket(src: SimControlResponsePacket) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_SimControlResponsePacket(writer, src)
    return writer.bytes()


def cdr_to_py_SimControlResponsePacket(cdr_payload) -> SimControlResponsePacket:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = SimControlResponsePacket()
    return cdr_body_to_py_SimControlResponsePacket(reader, dst)

File: pdu/javascript/hako_srv_msgs/pdu_cdr_conv_SimControlResponsePacket.js
import { SimControlResponsePacket } from './pdu_jstype_SimControlResponsePacket.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';
import { ServiceResponseHeader } from '../hako_srv_msgs/pdu_jstype_ServiceResponseHeader.js';
import { PduServiceResponseHeaderConverter } from '../hako_srv_msgs/pdu_cdr_conv_ServiceResponseHeader.js';
import { SimControlResponse } from '../hako_srv_msgs/pdu_jstype_SimControlResponse.js';
import { PduSimControlResponseConverter } from '../hako_srv_msgs/pdu_cdr_conv_SimControlResponse.js';


export class PduSimControlResponsePacketConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { SimControlResponsePacket } src
     */
    static to_cdr_body(writer, src) {
        PduServiceResponseHeaderConverter.to_cdr_body(writer, src.header);
        PduSimControlResponseConverter.to_cdr_body(writer, src.body);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { SimControlResponsePacket } dst
     * @returns { SimControlResponsePacket }
     */
    static cdr_body_to_js(reader, dst) {
        PduServiceResponseHeaderConverter.cdr_body_to_js(reader, dst.header);
        PduSimControlResponseConverter.cdr_body_to_js(reader, dst.body);
        return dst;
    }

    /**
     * @param { SimControlResponsePacket } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { SimControlResponsePacket }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new SimControlResponsePacket());
    }
}
